File: api-tester/src/main/java/com/revenuecat/apitester/java/VirtualCurrencyAPI.java
package com.revenuecat.apitester.java;

import androidx.annotation.OptIn;

import com.revenuecat.purchases.InternalRevenueCatAPI;
import com.revenuecat.purchases.virtualcurrencies.VirtualCurrency;

final class VirtualCurrencyAPI {
    static void check(final VirtualCurrency virtualCurrency) {
        final int balance = virtualCurrency.getBalance();
        final Integer balance2 = virtualCurrency.getBalance();
        final String name = virtualCurrency.getName();
        final String code = virtualCurrency.getCode();
        final String serverDescription = virtualCurrency.getServerDescription();
    }

    @OptIn(markerClass = InternalRevenueCatAPI.class)
    static void checkInternalRevenueCatAPIs() {
        VirtualCurrency virtualCurrencyWithServerDescription = new VirtualCurrency(
                100,
                "Gold",
                "GLD",
                "This is a test currency."
        );

        VirtualCurrency virtualCurrencyWithoutServerDescription = new VirtualCurrency(
                100,
                "Gold",
                "GLD",
                null
        );
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/VirtualCurrencyAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.InternalRevenueCatAPI
import com.revenuecat.purchases.virtualcurrencies.VirtualCurrency

@Suppress("unused", "UNUSED_VARIABLE")
private class VirtualCurrencyAPI {
    fun check(virtualCurrency: VirtualCurrency) {
        val balance: Int = virtualCurrency.balance
        val name: String = virtualCurrency.name
        val code: String = virtualCurrency.code
        val serverDescription: String? = virtualCurrency.serverDescription
    }

    @OptIn(InternalRevenueCatAPI::class)
    fun checkInternalRevenueCatAPIs() {
        val virtualCurrencyWithServerDescription = VirtualCurrency(
            balance = 100,
            name = "Gold",
            code = "GLD",
            serverDescription = "hello",
        )

        val virtualCurrencyWithoutServerDescription = VirtualCurrency(
            balance = 100,
            name = "Gold",
            code = "GLD",
            serverDescription = null,
        )
    }
}
